src/main/python/feature-extractor/server.py

```diff
@@ -14,93 +16,133 @@
 
 import pb.service_pb2 as service_pb2
 import pb.service_pb2_grpc as service_pb2_grpc
+# isn't used but re-exported so it can be used in tests
+from pb.service_pb2 import gopkg_dot_in_dot_bblfsh_dot_sdk_dot_v1_dot_uast_dot_generated__pb2 as uast_pb
 
 _ONE_DAY_IN_SECONDS = 60 * 60 * 24
 
+# keep extractors out of the Service class to be able to pickle them
+# return list instead of iterator for pickle also
+
+
+def _identifiers_extractor(uast, options):
+    return list(
+        IdentifiersBagExtractor(
+            docfreq_threshold=options.docfreqThreshold,
+            split_stem=options.splitStem,
+            weight=options.weight or 1).extract(uast))
+
+
+def _literals_extractor(uast, options):
+    return list(
+        LiteralsBagExtractor(
+            docfreq_threshold=options.docfreqThreshold,
+            weight=options.weight or 1).extract(uast))
+
+
+def _uast2seq_extractor(uast, options):
+    seq_len = list(options.seqLen) if options.seqLen else None
+
+    return list(
+        UastSeqBagExtractor(
+            docfreq_threshold=options.docfreqThreshold,
+            weight=options.weight or 1,
+            stride=options.stride or 1,
+            seq_len=seq_len or 5).extract(uast))
+
+
+def _graphlet_extractor(uast, options):
+    return list(
+        GraphletBagExtractor(
+            docfreq_threshold=options.docfreqThreshold,
+            weight=options.weight or 1).extract(uast))
+
+
+def _features_from_iter(f_iter):
+    return [service_pb2.Feature(name=f[0], weight=f[1]) for f in f_iter]
+
 
 class Service(service_pb2_grpc.FeatureExtractorServicer):
     """Feature Extractor Service"""
 
-    extractors_names = ["identifiers", "literals", "uast2seq", "graphlet"]
+    pool = None
+    extractors = {
+        "identifiers": _identifiers_extractor,
+        "literals": _literals_extractor,
+        "uast2seq": _uast2seq_extractor,
+        "graphlet": _graphlet_extractor,
+    }
+
+    def __init__(self, pool):
+        super(Service, self)
```

Comment:
Are you that this is working? This should be:

```suggestion
        super(Service, self).__init__()
```

Maybe it works because in the `service_pb2_grpc.FeatureExtractorServicer` class the init is actually a noop.

Reply:
🤦‍♀️ fixed